.github/workflows/docs_build.yml

```diff
@@ -0,0 +1,37 @@
+name: Sphinx docs to gh-pages
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
I don't think we want to build it on every commit to `main`. The docs should be up-to-date with the latest released version to avoid confusion. This can be done the same way as we do with the PyPI release, based on the tag of the commit.

Reply:
I don't think so, the github action should show us if the build fails and you can still allow manual build.